### DISCUSSION TITLE
How to handle connection errors to Kubernetes API server correctly?

### DISCUSSION THREAD

**Discussion by vvitkovsky** [dotnet/dotnet-operator-sdk#557] [Q&A]
Hello, 

I have a question related to handling connection errors to Kubernetes API server. 
If we start application with no connection to Kubernetes API server - we just have an UHE in 
``` await app.RunOperatorAsync(args); ```
e.g. System.Net.Http.HttpRequestException: 'No such host is known.'

And this just crash the whole app without any chance to restore connection later. 
Is there any workaround how we can handle this connection error and then restore connection when it will be possible? 

Best regards, 
Victor

**Comment by buehler** [comment]
Hey @vvitkovsky 

Well, in theory, the operator will run in Kubernetes itself. So, if there is no connection and the application crashes, Kubernetes will just restart the application. There is some retry logic directly built into Kubernetes (Pod Restart).

Currently, there is no real need for such a logic (IMO).
